server/chat-plugins/datasearch.ts

```diff
@@ -1549,81 +1550,90 @@ function runMovesearch(target: string, cmd: string, canAll: boolean, message: st
 
 	for (const alts of searches) {
 		if (alts.skip) continue;
-		for (const move in dex) {
+		for (const moveid in dex) {
+			const move = dex[moveid];
 			let matched = false;
 			if (Object.keys(alts.types).length) {
-				if (alts.types[dex[move].type]) continue;
-				if (Object.values(alts.types).includes(false) && alts.types[dex[move].type] !== false) continue;
+				if (alts.types[move.type]) continue;
+				if (Object.values(alts.types).includes(false) && alts.types[move.type] !== false) continue;
 			}
 
 			if (Object.keys(alts.categories).length) {
-				if (alts.categories[dex[move].category]) continue;
-				if (Object.values(alts.categories).includes(false) && alts.categories[dex[move].category] !== false) continue;
+				if (alts.categories[move.category]) continue;
+				if (Object.values(alts.categories).includes(false) && alts.categories[move.category] !== false) continue;
 			}
 
 			if (Object.keys(alts.contestTypes).length) {
-				if (alts.contestTypes[dex[move].contestType || 'Cool']) continue;
+				if (alts.contestTypes[move.contestType || 'Cool']) continue;
 				if (
 					Object.values(alts.contestTypes).includes(false) &&
-					alts.contestTypes[dex[move].contestType || 'Cool'] !== false
+					alts.contestTypes[move.contestType || 'Cool'] !== false
 				) continue;
 			}
 
 			if (Object.keys(alts.targets).length) {
-				if (alts.targets[dex[move].target]) continue;
-				if (Object.values(alts.targets).includes(false) && alts.targets[dex[move].target] !== false) continue;
+				if (alts.targets[move.target]) continue;
+				if (Object.values(alts.targets).includes(false) && alts.targets[move.target] !== false) continue;
 			}
 
 			for (const flag in alts.flags) {
 				if (flag === 'secondary') {
-					if ((!dex[move].secondary && !dex[move].secondaries) === !alts.flags[flag]) {
+					if ((!move.secondary && !move.secondaries) === !alts.flags[flag]) {
```

Comment:
```suggestion
					if (!(move.secondary || move.secondaries) === !alts.flags[flag]) {
```
Might as well refactor this while we're changing everything else.